lib/SPIRV/SPIRVReader.cpp

```diff
@@ -3284,89 +3284,17 @@ static bool transKernelArgTypeMedataFromString(LLVMContext *Ctx,
   return true;
 }
 
-bool SPIRVToLLVM::transKernelMetadata() {
+bool SPIRVToLLVM::transMetadata() {
   for (unsigned I = 0, E = BM->getNumFunctions(); I != E; ++I) {
     SPIRVFunction *BF = BM->getFunction(I);
     Function *F = static_cast<Function *>(getTranslatedValue(BF));
     assert(F && "Invalid translated function");
+
+    transOCLMetadata(BF);
```

Comment:
I suggest to move this call below the check for `spir` calling convention

Reply:
I think it is better to leave this function before the check. A couple of saved instructions don't worth breaking encapsulation